allocation.md

```diff
@@ -1,53 +1,406 @@
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    }
+  ]
+}
+```
+
+Note: querying for "3d" will likely return a range of four sets because the queried range will overlap with four precomputed 24-hour sets, each aligned to the configured timezone. For instance, querying "3d" on 2021/01/04T12:00:00 will return:
+- 2021/01/04 00:00:00 until 2021/01/04T12:00:00 (now)
+- 2021/01/03 00:00:00 until 2021/01/04 00:00:00
+- 2021/01/02 00:00:00 until 2021/01/03 00:00:00
+- 2021/01/01 00:00:00 until 2021/01/02 00:00:00
+
+See [Querying](#querying) for the full list of arguments and [Examples](#examples) for more example queries.
+
+## Allocation schema (version 1.76)
+Field | Description
+---: | :---
+name | Name of each relevant Kubernetes concept described by the allocation, delimited by slashes, e.g. "cluster/node/namespace/pod/container"
+properties | Map of name-to-value for all relevant property fields, including: `cluster`, `node`, `namespace`, `controller`, `controllerKind`, `pod`, `container`, `labels`, `annotation`, etc.
+window | Period of time over which the allocation is defined.
+start | Precise starting time of the allocation. By definition must be within the window.
+end | Precise ending time of the allocation. By definition must be within the window.
+minutes | Number of minutes running; i.e. the minutes from `start` until `end`.
+cpuCores | Average number of CPU cores allocated while running.
+cpuCoreRequestAverage | Average number of CPU cores requested while running.
+cpuCoreUsageAverage | Average number of CPU cores used while running.
+cpuCoreHours | Cumulative CPU core-hours allocated.
+cpuCost | Cumulative cost of allocated CPU core-hours.
+cpuEfficiency | Fraction of CPU request that was used.
+gpuHours | Cumulative GPU-hours allocated.
+gpuCost | Cumulative cost of allocated GPU-hours.
+networkCost | Cumulative cost of network usage.
+pvBytes | Average number of bytes of PersistentVolumes allocated while running.
+pvByteHours | Cumulative PersistenVolume byte-hours allocated.
+pvCost | Cumulative cost of allocated PersistentVolume byte-hours.
+ramBytes | Average number of RAM bytes allocated while running.
+ramByteRequestAverage | Average number of RAM bytes allocated while running.
+ramByteUsageAverage | Average number of RAM bytes used while running.
+ramByteHours | Cumulative RAM byte-hours allocated.
+ramCost | Cumulative cost of allocated RAM byte-hours.
+ramEfficiency | Fraction of RAM request that was used.
+sharedCost | Cumulative cost of shared resources, including: shared namespaces, shared labels, shared overhead.
+externalCost | Cumulative cost of external resources.
+totalCost | Total cumulative cost
+totalEfficiency | Cost-weighted average of `cpuEfficiency` and `ramEfficiency`
```

Comment:
Should we explain what "cost-weighted average" means?

Reply:
I didn't want to get too technical w/r/t equations, but we could try to provide examples somewhere. What would you recommend we add here as a description? The equation would be `((cpuEfficiency * cpuCost) + (ramEfficiency * ramCost)) / (cpuCost + ramCost)`, which isn't terribly long.

Reply:
I'm a fan of just including that equation. I think it can only help understanding.
```suggestion
totalEfficiency | Cost-weighted average of `cpuEfficiency` and `ramEfficiency`. In equation form: `((cpuEfficiency * cpuCost) + (ramEfficiency * ramCost)) / (cpuCost + ramCost)`
```